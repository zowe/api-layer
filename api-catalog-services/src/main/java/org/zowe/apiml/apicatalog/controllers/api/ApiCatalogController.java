/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

package org.zowe.apiml.apicatalog.controllers.api;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.zowe.apiml.apicatalog.exceptions.ContainerStatusRetrievalThrowable;
import org.zowe.apiml.apicatalog.model.APIContainer;
import org.zowe.apiml.apicatalog.services.cached.CachedApiDocService;
import org.zowe.apiml.apicatalog.services.cached.CachedProductFamilyService;
import org.zowe.apiml.message.log.ApimlLogger;
import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.stream.StreamSupport;

/**
 * Main API for handling requests from the API Catalog UI, routed through the gateway
 */
@Slf4j
@RestController
@RequestMapping("/")
@Tag(name = "API Catalog")
public class ApiCatalogController {

    private final CachedProductFamilyService cachedProductFamilyService;
    private final CachedApiDocService cachedApiDocService;

    @InjectApimlLogger
    private final ApimlLogger apimlLog = ApimlLogger.empty();

    /**
     * Create the controller and autowire in the repository services
     *
     * @param cachedProductFamilyService cached service for containers
     * @param cachedApiDocService        Cached state opf containers and services
     */
    @Autowired
    public ApiCatalogController(CachedProductFamilyService cachedProductFamilyService,
                                CachedApiDocService cachedApiDocService) {
        this.cachedProductFamilyService = cachedProductFamilyService;
        this.cachedApiDocService = cachedApiDocService;
    }


    /**
     * Get all containers
     *
     * @return a list of all containers
     */
    @GetMapping(value = "/containers", produces = MediaType.APPLICATION_JSON_VALUE)
    @Operation(summary = "Lists catalog dashboard tiles",
        description = "Returns a list of tiles including status and tile description",
        security = {
            @SecurityRequirement(name = "BasicAuthorization"), @SecurityRequirement(name = "CookieAuth")
        }
    )
    @ApiResponses(value = {
        @ApiResponse(responseCode = "200", description = "OK"),
        @ApiResponse(responseCode = "401", description = "Unauthorized"),
        @ApiResponse(responseCode = "403", description = "Forbidden"),
        @ApiResponse(responseCode = "404", description = "URI not found"),
        @ApiResponse(responseCode = "500", description = "An unexpected condition occurred")
    })
    public ResponseEntity<List<APIContainer>> getAllAPIContainers() throws ContainerStatusRetrievalThrowable {
        try {
            Iterable<APIContainer> allContainers = cachedProductFamilyService.getAllContainers();
            List<APIContainer> apiContainers = toList(allContainers);
            if (apiContainers == null || apiContainers.isEmpty()) {
                return new ResponseEntity<>(apiContainers, HttpStatus.NO_CONTENT);
            } else {
                // for each container, check the status of all it's services so it's overall status can be set here
                apiContainers.forEach(cachedProductFamilyService::calculateContainerServiceValues);
                return new ResponseEntity<>(apiContainers, HttpStatus.OK);
            }
        } catch (Exception e) {
            apimlLog.log("org.zowe.apiml.apicatalog.containerCouldNotBeRetrieved", e.getMessage());
            throw new ContainerStatusRetrievalThrowable(e);
        }
    }

    /**
     * Get all containers (and included services)
     *
     * @return a containers by id
     */
    @GetMapping(value = "/containers/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
    @Operation(summary = "Retrieves a specific dashboard tile information",
        description = "Returns information for a specific tile {id} including status and tile description",
        security = {
            @SecurityRequirement(name = "BasicAuthorization"), @SecurityRequirement(name = "CookieAuth")
        }
    )
    @ApiResponses(value = {
        @ApiResponse(responseCode = "200", description = "OK"),
        @ApiResponse(responseCode = "401", description = "Unauthorized"),
        @ApiResponse(responseCode = "403", description = "Forbidden"),
        @ApiResponse(responseCode = "404", description = "URI not found"),
        @ApiResponse(responseCode = "500", description = "An unexpected condition occurred")
    })
    public ResponseEntity<List<APIContainer>> getAPIContainerById(@PathVariable(value = "id") String id) throws ContainerStatusRetrievalThrowable {
        try {
            List<APIContainer> apiContainers = new ArrayList<>();
            APIContainer containerById = cachedProductFamilyService.getContainerById(id);
            if (containerById != null) {
                apiContainers.add(containerById);
            }
            if (!apiContainers.isEmpty()) {
                apiContainers.forEach(apiContainer -> {
                    // For this single container, check the status of all it's services so it's overall status can be set here
                    cachedProductFamilyService.calculateContainerServiceValues(apiContainer);
                    // add API Doc to the services to improve UI performance
                    setApiDocToService(apiContainer);
                });
            }
            return new ResponseEntity<>(apiContainers, HttpStatus.OK);
        } catch (Exception e) {
            apimlLog.log("org.zowe.apiml.apicatalog.containerCouldNotBeRetrieved", e.getMessage());
            throw new ContainerStatusRetrievalThrowable(e);
        }
    }

    private void setApiDocToService(APIContainer apiContainer) {
        apiContainer.getServices().forEach(apiService -> {
            // try the get the Api Doc for this service, if it fails for any reason then do not change the existing value
            // it may or may not be null
            String serviceId = apiService.getServiceId();
            try {
                String apiDoc = cachedApiDocService.getDefaultApiDocForService(serviceId);
                if (apiDoc != null) {
                    apiService.setApiDoc(apiDoc);
                }

                List<String> apiVersions = cachedApiDocService.getApiVersionsForService(serviceId);
                apiService.setApiVersions(apiVersions);

                String defaultApiVersion = cachedApiDocService.getDefaultApiVersionForService(serviceId);
                apiService.setDefaultApiVersion(defaultApiVersion);
            } catch (Exception e) {
                log.debug("An error occurred when trying to fetch ApiDoc for service: " + serviceId +
                    ", processing can continue but this service will not be able to display any Api Documentation.\n" +
                    "Error Message: " + e.getMessage());
            }
        });
    }

    /**
     * Convert an iterable to a list
     *
     * @param iterable the collection to convert
     * @param <T>      the type of the collection
     * @return a list
     */
    private <T> List<T> toList(final Iterable<T> iterable) {
        if (iterable == null) {
            return Collections.emptyList();
        }
        return StreamSupport.stream(iterable.spliterator(), false)
            .toList();
    }
}
