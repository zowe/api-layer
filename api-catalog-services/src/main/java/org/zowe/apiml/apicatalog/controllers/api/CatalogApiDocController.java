/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

package org.zowe.apiml.apicatalog.controllers.api;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.zowe.apiml.apicatalog.services.status.APIServiceStatusService;

/**
 * Main API for handling requests from the API Catalog UI, routed through the gateway
 */
@RestController
@RequestMapping("/apidoc")
@Tag(name = "API Documentation")
public class CatalogApiDocController {

    private final APIServiceStatusService apiServiceStatusService;

    /**
     * Create the controller and autowire in the repository services
     *
     * @param apiServiceStatusService repo service for registered services
     */
    @Autowired
    public CatalogApiDocController(APIServiceStatusService apiServiceStatusService) {
        this.apiServiceStatusService = apiServiceStatusService;
    }


    /**
     * Retrieve the api-doc info for this service
     *
     * @param serviceId the eureka id
     * @param apiId     the version of the api
     * @return api-doc info (as JSON)
     */
    @GetMapping(value = "/{serviceId}/{apiId}", produces = MediaType.APPLICATION_JSON_VALUE)
    @Operation(summary = "Retrieves the API documentation for a specific service version",
        description = "Returns the API documentation for a specific service {serviceId} and version {apiId}. When " +
            " the API documentation for the specified version is not found, the first discovered version will be used.",
        security = {
            @SecurityRequirement(name = "BasicAuthorization"), @SecurityRequirement(name = "CookieAuth")
        })
    @ApiResponses(value = {
        @ApiResponse(responseCode = "200", description = "OK"),
        @ApiResponse(responseCode = "401", description = "Unauthorized"),
        @ApiResponse(responseCode = "403", description = "Forbidden"),
        @ApiResponse(responseCode = "404", description = "URI not found"),
        @ApiResponse(responseCode = "500", description = "An unexpected condition occurred"),
    })
    public ResponseEntity<String> getApiDocInfo(
        @Parameter(name = "serviceId", description = "The unique identifier of the registered service", required = true, example = "apicatalog")
        @PathVariable(value = "serviceId") String serviceId,
        @Parameter(name = "apiId", description = "The API ID and version, separated by a space, of the API documentation", required = true, example = "zowe.apiml.apicatalog v1.0.0")
        @PathVariable(value = "apiId") String apiId) {
        return this.apiServiceStatusService.getServiceCachedApiDocInfo(serviceId, apiId);
    }

    /**
     * Retrieve the api-doc info for this service's default API
     *
     * @param serviceId the eureka id
     * @return api-doc info (as JSON)
     */
    @GetMapping(value = "/{serviceId}", produces = MediaType.APPLICATION_JSON_VALUE)
    @Operation(summary = "Retrieves the API documentation for the default service version",
        description = "Returns the API documentation for a specific service {serviceId} and its default version.",
        security = {
            @SecurityRequirement(name = "BasicAuthorization"), @SecurityRequirement(name = "CookieAuth")
        })
    @ApiResponses(value = {
        @ApiResponse(responseCode = "200", description = "OK"),
        @ApiResponse(responseCode = "401", description = "Unauthorized"),
        @ApiResponse(responseCode = "403", description = "Forbidden"),
        @ApiResponse(responseCode = "404", description = "URI not found"),
        @ApiResponse(responseCode = "500", description = "An unexpected condition occurred"),
    })
    public ResponseEntity<String> getDefaultApiDocInfo(
        @Parameter(name = "serviceId", description = "The unique identifier of the registered service", required = true, example = "apicatalog")
        @PathVariable(value = "serviceId") String serviceId) {
        return this.apiServiceStatusService.getServiceCachedDefaultApiDocInfo(serviceId);
    }

    @GetMapping(value = "/{serviceId}/{apiId1}/{apiId2}", produces = MediaType.TEXT_HTML_VALUE)
    @Operation(summary = "Retrieve diff of two api versions for a specific service",
        description = "Returns an HTML document which details the difference between two versions of a API service",
        security = {
            @SecurityRequirement(name = "BasicAuthorization"), @SecurityRequirement(name = "CookieAuth")
        })
    @ApiResponses(value = {
        @ApiResponse(responseCode = "200", description = "OK"),
        @ApiResponse(responseCode = "401", description = "Unauthorized"),
        @ApiResponse(responseCode = "403", description = "Forbidden"),
        @ApiResponse(responseCode = "404", description = "URI not found"),
        @ApiResponse(responseCode = "500", description = "An unexpected condition occurred")
    })
    public ResponseEntity<String> getApiDiff(
        @Parameter(name = "serviceId", description = "The unique identifier of the registered service", required = true, example = "apicatalog")
        @PathVariable(value = "serviceId") String serviceId,
        @Parameter(name = "apiId1", description = "The API ID and version, separated by a space, of the API documentation", required = true, example = "zowe.apiml.apicatalog v1.0.0")
        @PathVariable(value = "apiId1") String apiId1,
        @Parameter(name = "apiId2", description = "The API ID and version, separated by a space, of the API documentation", required = true, example = "zowe.apiml.apicatalog v2.0.0")
        @PathVariable(value = "apiId2") String apiId2) {
        return this.apiServiceStatusService.getApiDiffInfo(serviceId, apiId1, apiId2);
    }
}
