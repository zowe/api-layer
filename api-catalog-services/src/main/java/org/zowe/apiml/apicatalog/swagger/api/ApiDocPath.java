/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

package org.zowe.apiml.apicatalog.swagger.api;

import lombok.Getter;

import java.util.Map;
import java.util.Set;
import java.util.TreeMap;
import java.util.TreeSet;

@Getter
public class ApiDocPath<T> {
    private final Map<String, T> shortPaths = new TreeMap<>();
    private final Map<String, T> longPaths = new TreeMap<>();
    private final Set<String> prefixes = new TreeSet<>();

    public void addPrefix(String prefix) {
        getPrefixes().add(prefix);
    }

    public void addShortPath(String name, T path) {
        getShortPaths().put(name, path);
    }

    public void addLongPath(String name, T path) {
        getLongPaths().put(name, path);
    }
}
