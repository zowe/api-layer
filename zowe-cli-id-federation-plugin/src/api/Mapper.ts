/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import {ImperativeError, TextUtils} from "@zowe/imperative";
import {RacfCommands} from "./RacfCommands";
import {getAccount} from "./JobUtil";
import * as fs from "fs";
import {CsvParser, IIdentity} from "./CsvParser";
import {JclWriter} from "./JclWriter";
import {IHandlerResponseApi} from "@zowe/imperative/lib/cmd/src/doc/response/api/handler/IHandlerResponseApi";
import {Constants} from "./Constants";
import {TssCommands} from "./TssCommands";
import {Acf2Commands} from "./Acf2Commands";

export class Mapper {
    constructor(
        public file: string,
        public esm: string,
        public system: string,
        public registry: string,
        public response: IHandlerResponseApi) {
    }

    async map(): Promise<string> {
        const identities = new CsvParser(this.file,this.response).getIdentities();
        const commands = this.createSafCommands(identities);
        const jcl = await this.createJcl(commands);
        const fileMsg = this.system ? `_${this.system}` : "";
        const fileName = `idf_${this.esm}${fileMsg}.jcl`;
        fs.writeFileSync(fileName, jcl);

        const systemMsg = this.system ? ` on the system ${this.system}` : "";
        return `'${fileName}' was created. Review and submit this JCL${systemMsg}.`;
    }

    async createJcl(commands: string[]): Promise<string> {
        const jclTemplate = fs.readFileSync(`${__dirname}/templates/job.jcl`).toString();
        const account = await getAccount();
        const jclWriter = new JclWriter(1, 2);
        commands.forEach(c => jclWriter.add(c));
        return TextUtils.renderWithMustache(jclTemplate, {
            account: account,
            sysaff: this.system ? `\n/*JOBPARM SYSAFF=${this.system.toUpperCase()}` : "",
            system: this.system ? `system ${this.system}` : "a corresponding system",
            commands: jclWriter.getText()
        });
    }

    createSafCommands(identities: IIdentity[]): string[] {

        switch (this.esm.toLowerCase()) {
            case "racf": {
                return new RacfCommands(this.registry, identities, this.response).getCommands();
            }
            case "tss": {
                return new TssCommands(this.registry, identities, this.response).getCommands();
            }
            case "acf2": {
                return new Acf2Commands(this.registry, identities, this.response).getCommands();
            }
            default: {
                this.response.data.setExitCode(Constants.FATAL_CODE);
                const msg = `Unsupported ESM "${this.esm}".` +
                    `Id Federation Plugin supports only the following security systems: RACF, TSS, and ACF2.`;
                throw new ImperativeError({msg});
            }
        }
    }
}
