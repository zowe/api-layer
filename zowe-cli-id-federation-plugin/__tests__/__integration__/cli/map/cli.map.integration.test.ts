/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import * as fs from "fs";
import * as path from "path";
import {ITestEnvironment, runCliScript, stripProfileDeprecationMessages, TestEnvironment} from "@zowe/cli-test-utils";
import {ITestPropertiesSchema} from "../../../__src__/environment/doc/ITestPropertiesSchema";
import {Constants} from "../../../../src/api/Constants";
import {expect, describe, it, beforeAll, afterAll} from '@jest/globals';

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;

const testCsv = "users.csv";

describe("id-federation map command integration tests", () => {

    let csv: string;

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "map_command",
            skipProperties: true
        });

        csv = path.join(TEST_ENVIRONMENT.workingDir, testCsv);
        fs.copyFileSync(path.join(__dirname, "__resources__", testCsv), csv);
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should fail when options are not passed", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_error_handler.sh", TEST_ENVIRONMENT, [csv]);

        expect(response.status).toBe(Constants.ZOWE_ERROR_CODE);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should fail when arguments are not valid", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_error_handler_invalid_argument.sh", TEST_ENVIRONMENT, [csv]);

        expect(response.status).toBe(Constants.ZOWE_ERROR_CODE);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should print the successful creation message with passed args for RACF", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_with_passed_args.sh", TEST_ENVIRONMENT,
            [`${TEST_ENVIRONMENT.workingDir}/users.csv`, "RACF", "TST1", "ldap://12.34.56.78:910"]);

        expect(response.status).toBe(Constants.OKAY_CODE);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();

        const jcl = fs.readFileSync("idf_RACF_TST1.jcl").toString();
        expect(jcl).toMatchSnapshot();
    });

    it("should print the successful creation message with passed args for TSS", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_with_passed_args.sh", TEST_ENVIRONMENT,
            [`${TEST_ENVIRONMENT.workingDir}/users.csv`, "TSS", "TST1", "ldap://12.34.56.78:910"]);

        expect(response.status).toBe(Constants.OKAY_CODE);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();

        const jcl = fs.readFileSync("idf_TSS_TST1.jcl").toString();
        expect(jcl).toMatchSnapshot();
    });

    it("should print the successful creation message with passed args for ACF2", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_with_passed_args.sh", TEST_ENVIRONMENT,
            [`${TEST_ENVIRONMENT.workingDir}/users.csv`, "ACF2", "TST1", "ldap://12.34.56.78:910"]);

        expect(response.status).toBe(Constants.OKAY_CODE);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();

        const jcl = fs.readFileSync("idf_ACF2_TST1.jcl").toString();
        expect(jcl).toMatchSnapshot();
    });

    it("should return command error in case of missing arguments", () => {
        const response = runCliScript(__dirname + "/__scripts__/map_with_passed_args.sh", TEST_ENVIRONMENT,
            [csv, "TSS", "", "ldap://12.34.56.78:910"]);

        expect(response.status).toBe(Constants.ZOWE_ERROR_CODE);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
    });

});
