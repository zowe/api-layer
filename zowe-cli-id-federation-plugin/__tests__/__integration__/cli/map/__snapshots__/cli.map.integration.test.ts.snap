// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`id-federation map command integration tests should fail when arguments are not valid 1`] = `
"Unable to perform this operation due to the following problem.
Unknown argument: invalidarg
Command failed due to improper syntax
Did you mean: plugins validate?

Command entered: "idf invalidarg"
Use "zowe idf --help" to view groups, commands, and options.

Response From Service
Error: Unknown argument: invalidarg

Diagnostic Information
Unknown argument: invalidarg
"
`;

exports[`id-federation map command integration tests should fail when arguments are not valid 2`] = `""`;

exports[`id-federation map command integration tests should fail when options are not passed 1`] = `
"
Syntax Error:
Missing Required Option:
--esm (-e)

Option Description:
The ESM product on the target system

Syntax Error:
Missing Required Option:
--registry (-r)

Option Description:
The distributed identities registry (e.g., ldaps://enterprise.com,
ldap://12.34.56.78:389)

Use "zowe idf map --help" to view command description, usage, and options.
"
`;

exports[`id-federation map command integration tests should fail when options are not passed 2`] = `""`;

exports[`id-federation map command integration tests should print the successful creation message with passed args for ACF2 1`] = `
"'idf_ACF2_TST1.jcl' was created. Review and submit this JCL on the system TST1.
"
`;

exports[`id-federation map command integration tests should print the successful creation message with passed args for ACF2 2`] = `
"//IDF JOB (account),CLASS=A,MSGCLASS=X
/*JOBPARM SYSAFF=TST1
//*-------------------------------------------------------------------
//*
//* Zowe CLI ID Federation Plugin
//*
//* This JCL can be used for mapping from a distributed identity to
//* a mainframe identity
//*
//* 1) Validate the job header and make all necessary changes
//*    according to your system environment.
//*
//* 2) Run the JCL on system TST1.
//*
//* Note(s):
//*
//* 1. THE USER ID THAT RUNS THIS JOB MUST HAVE SUFFICIENT AUTHORITY
//*    TO ALTER SECURITY DEFINITIONS
//*
//*-------------------------------------------------------------------
//RUN      EXEC PGM=IKJEFT01,REGION=0M
//SYSTSPRT DD SYSOUT=*
//SYSTSIN     DD DATA,SYMBOLS=JCLONLY
 ACF
 SET PROFILE(USER) DIVISION(IDMAP)
 INSERT mf_jir.ZWEDNMAP IDMAPDN(dist_jirka) -
  IDMAPRN(ldap://12.34.56.78:910) IDLABEL(Jirka)
 INSERT mf_lena.ZWEDNMAP IDMAPDN(dist_lena) -
  IDMAPRN(ldap://12.34.56.78:910) IDLABEL(Lena)
 INSERT mf_pab.ZWEDNMAP IDMAPDN(dist_pablo) -
  IDMAPRN(ldap://12.34.56.78:910) IDLABEL(Pablo)
 INSERT mf_name.ZWEDNMAP IDMAPDN(Dist -
  naaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaame ok) -
  IDMAPRN(ldap://12.34.56.78:910) IDLABEL(Name)
 
 F ACF2,REBUILD(USR),CLASS(P),DIVISION(IDMAP)
 END
/*
"
`;

exports[`id-federation map command integration tests should print the successful creation message with passed args for RACF 1`] = `
"'idf_RACF_TST1.jcl' was created. Review and submit this JCL on the system TST1.
"
`;

exports[`id-federation map command integration tests should print the successful creation message with passed args for RACF 2`] = `
"//IDF JOB (account),CLASS=A,MSGCLASS=X
/*JOBPARM SYSAFF=TST1
//*-------------------------------------------------------------------
//*
//* Zowe CLI ID Federation Plugin
//*
//* This JCL can be used for mapping from a distributed identity to
//* a mainframe identity
//*
//* 1) Validate the job header and make all necessary changes
//*    according to your system environment.
//*
//* 2) Run the JCL on system TST1.
//*
//* Note(s):
//*
//* 1. THE USER ID THAT RUNS THIS JOB MUST HAVE SUFFICIENT AUTHORITY
//*    TO ALTER SECURITY DEFINITIONS
//*
//*-------------------------------------------------------------------
//RUN      EXEC PGM=IKJEFT01,REGION=0M
//SYSTSPRT DD SYSOUT=*
//SYSTSIN     DD DATA,SYMBOLS=JCLONLY
 RACMAP ID(mf_jir) MAP USERDIDFILTER(NAME('dist_jirka')) -
  REGISTRY(NAME('ldap://12.34.56.78:910')) WITHLABEL('Jirka')
 RACMAP ID(mf_lena) MAP USERDIDFILTER(NAME('dist_lena')) -
  REGISTRY(NAME('ldap://12.34.56.78:910')) WITHLABEL('Lena')
 RACMAP ID(mf_pab) MAP USERDIDFILTER(NAME('dist_pablo')) -
  REGISTRY(NAME('ldap://12.34.56.78:910')) WITHLABEL('Pablo')
 RACMAP ID(mf_name) MAP -
  USERDIDFILTER(NAME('Dist naaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaame ok')) REGISTRY(NAME('ldap://12.34.56.78:910')) -
  WITHLABEL('Name')
 
 SETROPTS RACLIST(IDIDMAP) REFRESH
/*
"
`;

exports[`id-federation map command integration tests should print the successful creation message with passed args for TSS 1`] = `
"'idf_TSS_TST1.jcl' was created. Review and submit this JCL on the system TST1.
"
`;

exports[`id-federation map command integration tests should print the successful creation message with passed args for TSS 2`] = `
"//IDF JOB (account),CLASS=A,MSGCLASS=X
/*JOBPARM SYSAFF=TST1
//*-------------------------------------------------------------------
//*
//* Zowe CLI ID Federation Plugin
//*
//* This JCL can be used for mapping from a distributed identity to
//* a mainframe identity
//*
//* 1) Validate the job header and make all necessary changes
//*    according to your system environment.
//*
//* 2) Run the JCL on system TST1.
//*
//* Note(s):
//*
//* 1. THE USER ID THAT RUNS THIS JOB MUST HAVE SUFFICIENT AUTHORITY
//*    TO ALTER SECURITY DEFINITIONS
//*
//*-------------------------------------------------------------------
//RUN      EXEC PGM=IKJEFT01,REGION=0M
//SYSTSPRT DD SYSOUT=*
//SYSTSIN     DD DATA,SYMBOLS=JCLONLY
 TSS ADD(mf_jir) IDMAP(ZWEDNMAP) IDMAPDN('dist_jirka') -
  IDMAPRN('ldap://12.34.56.78:910') IDLABEL('Jirka')
 TSS ADD(mf_lena) IDMAP(ZWEDNMAP) IDMAPDN('dist_lena') -
  IDMAPRN('ldap://12.34.56.78:910') IDLABEL('Lena')
 TSS ADD(mf_pab) IDMAP(ZWEDNMAP) IDMAPDN('dist_pablo') -
  IDMAPRN('ldap://12.34.56.78:910') IDLABEL('Pablo')
 TSS ADD(mf_name) IDMAP(ZWEDNMAP) -
  IDMAPDN('Dist naaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaame o-
k') IDMAPRN('ldap://12.34.56.78:910') IDLABEL('Name')
/*
"
`;

exports[`id-federation map command integration tests should return command error in case of missing arguments 1`] = `
"
Syntax Error:
No value specified for option:
--system

This option requires a value of type:
string

Option Description:
The target JES system on which the command will be executed

Syntax Error:
Invalid value length for option:
--system

You specified a string of length 0:


The length must be between 1 and 8 (inclusive)

Use "zowe idf map --help" to view command description, usage, and options.
"
`;

exports[`id-federation map command integration tests should return command error in case of missing arguments 2`] = `""`;
